index.bs

```diff
@@ -801,31 +822,67 @@ partial interface MLContext {
-    1. For each |key| -> |value| of |outputs|:
+    1. For each |key| -> |value| of |transferredOutputs|:
         1. Issue a compute request for output of |graph|.{{MLGraph/[[implementation]]}} that is associated with |key|.
         1. Wait for the compute request to be completed.
         1. If there is an error returned by |graph|.{{MLGraph/[[implementation]]}}, then:
             1. [=reject=] |promise| with an "{{OperationError}}" {{DOMException}} and stop.
         1. Else:
             1. Let |outputTensor| be the output tensor returned by |graph|.{{MLGraph/[[implementation]]}}.
             1. Let |outputDesc| be |graph|.{{MLGraph/[[outputDescriptors]]}}[|key|].
-            1. If the data type of |outputTensor| doesn't match the [=element type=] of {{ArrayBufferView}} |value|, then throw a "{{DataError}}" {{DOMException}} and stop.
+            1. If the data type of |outputTensor| doesn't match the [=element type=] of {{ArrayBufferView}} |value|, then:
+                1. [=reject=] |promise| with a "{{DataError}}" {{DOMException}} and stop.
             1. If the byte length of |outputTensor| is not equal to [=byte length=] of |outputDesc|, then:
                 1. [=reject=] |promise| with an "{{OperationError}}" {{DOMException}} and stop.
             1. Else:
                 1. Set the values of |value| to the values of |outputTensor|.
-            1. If all compute requests are completed, [=Resolve=] |promise| and stop.
+            1. If all compute requests are completed, then:
+                1. Let |result| be a new {{ComputeResult}}.
+                1. Set |result|.{{ComputeResult/inputs}} to |transferredInputs|.
+                1. Set |result|.{{ComputeResult/outputs}} to |transferredOutputs|.
+                1. [=Resolve=] |promise| with |result| and stop.
     1. Return |promise|.
 </div>
 
+#### Examples #### {#api-mlcontext-async-execution-examples}
+
+<div class="example">
+The following code showcases the asynchronous computation.
+<pre highlight="js">
+const operandType = {type: 'float32', dimensions: [2, 2]};
+const context = await navigator.ml.createContext();
+const builder = new MLGraphBuilder(context);
+// 1. Create a computational graph 'C = 0.2 * A + B'.
+const constant = builder.constant(0.2);
+const A = builder.input('A', operandType);
+const B = builder.input('B', operandType);
+const C = builder.add(builder.mul(A, constant), B);
+// 2. Compile it into an executable.
+const graph = await builder.build({'C': C});
+// 3. Bind inputs to the graph and execute for the result.
+const bufferA = new Float32Array(4).fill(1.0);
+const bufferB = new Float32Array(4).fill(0.8);
+const bufferC = new Float32Array(4);
+const inputs = {'A': bufferA, 'B': bufferB};
+const outputs = {'C': bufferC};
+const result = await context.compute(graph, inputs, outputs);
+// The computed result of [[1, 1], [1, 1]] is in the buffer associated with
+// the output operand.
+console.log('Output value: ' + result.C);
```

Comment:
Should this be `result.outputs.C`?

Reply:
Sorry, my typo. Fixed in the latest commit.